File: src/main/java/app/askresume/web/google/GoogleTokenClient.java
package app.askresume.web.google;


import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(url = "https://oauth2.googleapis.com", name = "googleTokenClient")
public interface GoogleTokenClient {

    @PostMapping(value = "/token", consumes = "application/json")
    Object requestGoogleToken(@RequestBody GoogleTokenDto.Request request);


}

File: src/main/kotlin/app/askresume/web/google/GoogleTokenClient.kt
package app.askresume.web.google

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody


@FeignClient(url = "\${oauth.google.client-url}", name = "googleTokenClient")
interface GoogleTokenClient {

    @PostMapping(value = ["/token"], consumes = ["application/json"])
    fun requestGoogleToken(@RequestBody request: GoogleTokenDto.Request): Any

}
